serveradmin/serverdb/forms.py

```diff
@@ -0,0 +1,42 @@
+from django import forms
+from django.core.exceptions import ValidationError
+from django.utils.translation import gettext as _
+
+
+class ServertypeAdminForm(forms.ModelForm):
+    def clean(self):
+        if self.cleaned_data.get('ip_addr_type') == 'null':
+            supernet_and_required = [
+                a.attribute_id for a in
+                self.instance.attributes.filter(
+                    required=True, attribute__type='supernet')]
+            if supernet_and_required:
+                raise ValidationError(_(
+                        'ip_addr_type for Servertype must not be "null" '
+                        'if any attribute of type supernet is required '
+                        'but %(attrs)s is/are!'
+                    ),
+                    code='invalid',
+                    params={
+                        'attrs': ','.join(supernet_and_required)
+                    })
+        super().clean()
+
+
+class ServertypeAttributeAdminForm(forms.ModelForm):
+    def clean(self):
+        if (
+            self.instance.attribute.type == 'supernet' and
```

Comment:
I would suggest to introduce another variable with a meaningful name just like above. So, it is easier to read the purpose of the condition statement.

Reply:
fair enough.